ionoscloud/resource_dbaas_pgsql_user.go

```diff
@@ -0,0 +1,180 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	pgsql "github.com/ionos-cloud/sdk-go-dbaas-postgres"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/dbaas"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"strings"
+)
+
+func resourceDbaasPgSqlUser() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDbaasPgSqlUserCreate,
+		UpdateContext: resourceDbaasPgSqlUserUpdate,
+		ReadContext:   resourceDbaasPgSqlUserRead,
+		DeleteContext: resourceDbaaSPgSqlUserDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDbaasPgSqlUserImporter,
+		},
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"is_system_user": {
+				Type:        schema.TypeBool,
+				Description: "Describes whether this user is a system user or not. A system user cannot be updated or deleted.",
+				Computed:    true,
+				Optional:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceDbaasPgSqlUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).PsqlClient
+
+	clusterId := d.Get("cluster_id").(string)
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+
+	request := pgsql.User{
+		Properties: &pgsql.UserProperties{},
+	}
+	request.Properties.Username = &username
+	request.Properties.Password = &password
+
+	if isSystemUser, ok := d.GetOk("is_system_user"); ok {
+		isSystemUserValue := isSystemUser.(bool)
+		request.Properties.System = &isSystemUserValue
+	}
+
+	user, _, err := client.CreateUser(ctx, clusterId, request)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while adding an user to the PgSql cluster with ID: %s, error: %w", clusterId, err))
+	}
+	d.SetId(*user.Id)
+	// Wait for the cluster to be ready again (when creating/updating the user, the cluster enters
+	// 'BUSY' state).
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error while waiting for PgSql cluster with ID: %s to be ready, error: %w", clusterId, err))
```

Comment:
let's add the user.Id her, otherwise it will seem like we're waiting for the cluster

Reply:
let' still add `error while waiting for PgSql cluster with ID: %s to be ready, after creating user with username %s"